pipeline/inferrer/feature_training/src/elastic.py

```diff
@@ -1,46 +1,62 @@
 import os
-
+import random
 import numpy as np
 from elasticsearch import Elasticsearch, helpers
+from .logging import get_logger
 
+logger = get_logger(__name__)
 
-def get_all_ids(es_client, index_name):
-    response = helpers.scan(
-        client=es_client,
-        index=index_name,
-        query={"query": {"match_all": {}}, "stored_fields": []},
-    )
-
-    return [hit["_id"] for hit in list(response)]
 
+def get_documents_from_es_in_batches(es_client, index_name, ids, batch_size=5000):
```

Comment:
```suggestion
def get_all_documents_from_es(es_client, index_name, ids, batch_size=5000):
```

The caller only cares that you get all the documents; that it's happening in batches is mostly irrelevant to them.

Reply:
Would it be worth allowing the caller to specify extra fields on the body here? You're only using two fields on the returned documents; I don't know how many fields there are but it might be a tidy saving.

Reply:
It actually only returns the 2 fields we want because of field-level permissions on the kibana role, but I may add this anyway